core/Cargo.toml

```
@@ -163,6 +166,7 @@ http = "0.2.5"
 hyper = "0.14"
 lazy-regex = { version = "2.5.0", optional = true }
 log = "0.4"
+madsim = {version = "0.2.19", optional = true }
```

use the latest version as it contains various fixes.
```suggestion
madsim = { version = "0.2.21", optional = true }
```